lambda/orc_ingestion.py

```diff
@@ -15,116 +16,88 @@
-
-    for nhs_number in nhs_number_list:
-        trace_id = new_uuid()
-        all_trace_ids.append(trace_id)
-        logger.info(f"Assigned a new traceId: {trace_id}")
-
-        suspension_message_json = build_suspension_message(
-            nhs_number=nhs_number,
-            previous_ods_code=previous_ods_code,
-            nems_message_id=trace_id,
-        )
-        send_message_with_trace_id(
-            message_body=suspension_message_json,
-            queue_url=suspension_queue_url,
-            trace_id=trace_id,
-        )
-        logger.info(
-            f"sent message for a patient to queue with traceId: {trace_id}"
-        )
-
-    logger.info("Here are all the trace ids that related to this ingest:")
-    logger.info(str(all_trace_ids))
-
-    return {"statusCode": 200, "body": str(all_trace_ids)}
+class DuplicateNhsNumberException(Exception):
+    pass
 
 
-def get_nhs_number_list_from_s3(filename: str, bucket_name: str) -> list:
-    response = s3_client.get_object(
-        Bucket=bucket_name,
-        Key=filename
-    )
+class InvalidFileFormatException(Exception):
+    pass
 
-    file_lines = response["Body"].readlines()
 
-    if len(file_lines) > 1:
-        raise InvalidFileFormatException(
-            'All NHS numbers must be contained on a single line, seperated by commas.'
-        )
+class InvalidNhsNumberException(Exception):
+    pass
 
-    nhs_numbers_str = file_lines[0].decode('utf-8')
 
-    only_nums_and_commas_bool = bool(re.match('^[0-9,]+$', nhs_numbers_str))
+class NHSNumberValidator:
+    @staticmethod
+    def validate(nhs_numbers: str) -> List[str]:
+        if not bool(re.match('^[0-9,]+$', nhs_numbers)):
+            raise InvalidFileFormatException('File should only contain numbers and commas.')
 
-    if not only_nums_and_commas_bool:
-        raise InvalidFileFormatException(
-            'File should only contain numbers and commas on a single line.'
-        )
+        nhs_number_list = nhs_numbers.split(',')
+        if not all(len(x) == 10 for x in nhs_number_list):
+            raise InvalidNhsNumberException('All NHS numbers must be 10 digits long and there must be no trailing commas.')
 
-    nhs_number_list = nhs_numbers_str.split(',')
-    all_length_10_bool = all(len(x) == 10 for x in nhs_number_list)
+        if len(nhs_number_list) != len(set(nhs_number_list)):
+            raise DuplicateNhsNumberException('Duplicate NHS numbers found.')
 
-    if not all_length_10_bool:
-        raise InvalidNhsNumberException(
-            'All NHS numbers must be 10 digits long and there must be no trailing commas.'
-        )
+        return nhs_number_list
 
-    return nhs_number_list
 
+class S3FileReader:
+    @staticmethod
+    def read(filename, bucket_name) -> str:
+        response = s3_client.get_object(Bucket=bucket_name, Key=filename)
+        file_lines = response["Body"].readlines()
 
-def new_uuid() -> str:
-    return str(uuid.uuid4())
+        if len(file_lines) > 1:
```

Comment:
bit pedantic so feel free to ignore:

I would separate this validation out into a separate function under the class and then calls Class.read() and then class.validate_one_line()